### DISCUSSION TITLE
micropython core esp32 c3

### DISCUSSION THREAD

**Discussion by icelon** [micropython/micropython#11459] [General]
hello ||
, I have acquired a board type esp32 c3 core (attached photo) and I cannot make it work, the whole process works correctly, only when I add it to the ide thonny it does not detect the interpreter, I have used the commands that are on the page where to download the firmware, I would appreciate any help on how to make it work
![luatos-esp32c3](https://github.com/micropython/micropython/assets/2294373/615d2d8e-ebe2-48f1-8c08-6a9016b80c6f)

**Comment by icelon** [comment]
process 

` esptool.py --chip esp32c3 --port /dev/ttyACM0 erase_flashesptool.py v4.5.1
Serial port /dev/ttyACM0
Connecting.....
Chip is ESP32-C3 (revision v0.3)
Features: WiFi, BLE
Crystal is 40MHz
MAC: 60:55:f9:7b:55:b8
Uploading stub...
Running stub...
Stub running...
Erasing flash (this may take a while)...
Chip erase completed successfully in 2.1s
Hard resetting via RTS pin...
 esptool.py --chip esp32c3 --port /dev/ttyACM0 --baud 460800 write_flash -z 0x0 esp32c3-usb-20230426-v1.20.0.bin 
esptool.py v4.5.1
Serial port /dev/ttyACM0
Connecting....
Chip is ESP32-C3 (revision v0.3)
Features: WiFi, BLE
Crystal is 40MHz
MAC: 60:55:f9:7b:55:b8
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Configuring flash size...
Flash will be erased from 0x00000000 to 0x00164fff...
Compressed 1461232 bytes to 891797...
Wrote 1461232 bytes (891797 compressed) at 0x00000000 in 20.9 seconds (effective 558.9 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...

 rshell 
Welcome to rshell. Use Control-D (or the exit command) to exit rshell.

No MicroPython boards connected - use the connect command to add `one`

**Reply by robert-hh** [reply]
The picture of the board is not very clear, but there might be a USB/UART bridge there. You could try the UART version of the firmware from https://micropython.org/download/esp32c3/

**Comment by icelon** [comment]
https://templates.blakadder.com/luatos_CORE-ESP32.html

**Reply by icelon** [reply]
https://wiki.luatos.com/_static/bom/esp32c3.html

**Comment by robert-hh** [comment]
Yes, it looks like a CP2102 on theboard. Did you try to install the other firmware, the non-USB version?

**Reply by gitmun** [reply]
yes, and it still doesn't work

**Comment by icelon** [comment]
ummm it's a CH343P .... I'll look around to see but it doesn't go with any compilation of esp32-c3

**Reply by robert-hh** [reply]
Just another USB/UART bridge. It does not require a special compile variant of the ESP32C3, besides serial REPL. I do not expect it to need a specific driver for Linux. When the board is connected, does it register as /dev/ttyACMx or /dev/ttyUSBx? Your log shows /dev/ttyACM0, which is unusual for a USB/UART bridge,

**Reply by gitmun** [reply]
Yes is ttyacm0...

**Reply by robert-hh** [reply]
That's a little bit surprising. The boards with CH340 I have come up as /dev/ttyUSBx. But the CH343 may register differently, and the firmware upload works apparently with that device. Do you use plain Linux of Windows-WSL?
What happens if you use `mpremote` or `picocom -b 115200 /dev/ttyACM0` as terminal emulator instead of rshell?

**Comment by icelon** [comment]
picocom -b 115200 /dev/ttyACM0 
picocom v3.1

port is        : /dev/ttyACM0
flowcontrol    : none
baudrate is    : 115200
parity is      : none
databits are   : 8
stopbits are   : 1
escape is      : C-a
local echo is  : no
noinit is      : no
noreset is     : no
hangup is      : no
nolock is      : no
send_cmd is    : sz -vv
receive_cmd is : rz -vv -E
imap is        : 
omap is        : 
emap is        : crcrlf,delbs,
logfile is     : none
initstring     : none
exit_after is  : not set
exit is        : no

Type [C-a] [C-h] to see available commands
Terminal ready
MicroPython v1.20.0 on 2023-04-26; ESP32C3 module with ESP32C3
Type "help()" for more information.
>>> help ('modules')
__main__          gc                ubinascii         urandom
_boot             inisetup          ubluetooth        ure
_onewire          math              ucollections      urequests
_thread           micropython       ucryptolib        uselect
_uasyncio         mip/__init__      uctypes           usocket
_webrepl          neopixel          uerrno            ussl
apa106            network           uhashlib          ustruct
btree             ntptime           uheapq            usys
builtins          onewire           uio               utime
cmath             uarray            ujson             utimeq
dht               uasyncio/__init__ umachine          uwebsocket
ds18x20           uasyncio/core     umqtt/robust      uzlib
esp               uasyncio/event    umqtt/simple      webrepl
esp32             uasyncio/funcs    uos               webrepl_setup
flashbdev         uasyncio/lock     uplatform
framebuf          uasyncio/stream   upysh
Plus any modules on the filesystem
>>>

**Reply by robert-hh** [reply]
So it works!
